src/main/java/org/apache/sysml/api/mlcontext/MLResults.java

```diff
@@ -115,7 +120,21 @@ public Data getData(String outputName) {
 	 */
 	public MatrixObject getMatrixObject(String outputName) {
 		Data data = getData(outputName);
-		if (!(data instanceof MatrixObject)) {
+		if(data instanceof ScalarObject) {
```

Comment:
This unifies returning scalar as well as matrix and simplifies the python interface. Currently, if the registered output variable turns out to be scalar, then this change will allow it to be returned as 1 X 1 matrix.